xen/drivers/passthrough/iommu.c

```diff
@@ -273,21 +290,37 @@ int iommu_map_page(struct domain *d, unsigned long gfn, unsigned long mfn,
     return rc;
 }
 
-int iommu_unmap_page(struct domain *d, unsigned long gfn)
+int iommu_unmap_pages(struct domain *d, unsigned long gfn,
+                      unsigned long page_count)
 {
     const struct domain_iommu *hd = dom_iommu(d);
-    int rc;
+    int ret, rc = 0;
```

Comment:
Any reason to crash? Driver must support at least one of these callbacks (unmap_page/unmap_pages). So, rc won't be rewritten only in one case when unmap_page is used and page_count = 0.